src/repositories/log_interface.py

```diff
@@ -8,10 +9,9 @@ class LogInterface(ABC):
     """
 
     @abstractmethod
-    def get_dataframe(self, project: str) -> pd.DataFrame:
+    def get_dataframe(self, project: str) -> Generator[pd.DataFrame, None, None]:
         """
-        Get a dataframe containing header/metadata of all
-        logs at given project.
+        Generate dataframes containing header/metadata of all logs at given project.
```

Comment:
```suggestion
        Generate dataframes containing header of all logs at given project.
```

(ref https://github.com/equinor/recall-dsis-client/issues/4 )